### DISCUSSION TITLE
Making sensors optional

### DISCUSSION THREAD

**Discussion by deimi** [rbaron/b-parasite#205] [Ideas]
Today I ordered a bunch of PCBs without the ambient light and Sensirion chip as I don't need this features and it makes the hardware cheaper.  
In order to have a clean firmware without any errors, I will disable those sensors in the firmware as well. Either completely remove the parts or make it configurable via Kconfig.

@rbaron are you interested in such a upstream contribution? I'm asking to know if it makes sense to make in a clean way and submit a PR or just make dirty workaround in my personal fork.

**Comment by rbaron** [comment]
@deimi yes, that's something on my TODO list. I'm happy to review the PR. Some pointers of the implementation I have in mind:
- Sensors flag-gated at the `prstlib` level
- Kconfig options like `CONFIG_PRSTLIB_USE_{X, Y}`, defaults to `y`
- Ideally the `ble`, `zigbee` samples should not advertise values for disabled sensors

**Reply by deimi** [reply]
Ok, sounds good.  
I haven't had a detailed look yet, but do you think that zigbee2mqtt (with a single b-parasite.js) will be able to handle flawlessly different kind of b-parasites when they advertise different kind of values?

**Reply by rbaron** [reply]
That's a good question. Our custom converter [b-parasite.js](https://github.com/rbaron/b-parasite/blob/main/code/nrf-connect/samples/zigbee/b-parasite.js) "sets up" all sensors. I actually don't know what would happen if we only had clusters for some of them on the device. I'd guess they would be set up in HA but be unavailable/unknown. We can always disable them manually. I can't find it now, but a while ago I read somewhere that z2m introduced something more automatic for the converters, which would let us get rid of the custom converter, and then I would expect it to work automatically with a subset of sensors.

EDIT: found it: 
- https://github.com/Koenkk/zigbee-herdsman-converters/pull/6692
- https://github.com/Koenkk/zigbee2mqtt/pull/20358

It would be cool if we got this to work (I haven't tested it).


With ZHA I expect it to work out of the box, as long as we flag-gate the cluster definitions in the zigbee sample, because it automatically detects the exposed clusters and turn them into HA entities.

**Reply by deimi** [reply]
Ok, I will take a look

**Comment by flod-1** [comment]
Are there any updates? I also want to order, but I don't need the light sensor at all, but I will keep the SHTC3 sensor.

**Comment by rbaron** [comment]
@flod-1 for now you should be able to ignore the sensor values in the UI. Side note: I also don't use the light sensor on my boards, but I still populate them, because the marginal cost for the phototransistor is so low.
